src/coord/timestamp.rs

```diff
@@ -777,122 +1166,165 @@ impl Timestamper {
             config.set(k, v);
         }
 
-        let k_consumer: BaseConsumer = config.create().expect("Failed to create Kakfa consumer");
-        RtKafkaConnector {
-            consumer: k_consumer,
-            topic: kc.topic,
+        match config.create() {
+            Ok(consumer) => Some(RtKafkaConnector {
+                consumer,
+                topic: kc.topic,
+            }),
+            Err(e) => {
+                error!("Failed to create Kafka Consumer {}", e);
+                None
+            }
         }
     }
 
+    #[allow(clippy::redundant_clone)]
+    fn create_rt_ocf_connector(
+        &self,
+        _id: SourceInstanceId,
+        fc: FileSourceConnector,
+    ) -> Option<RtFileConnector<avro::types::Value>> {
+        let ctor = move |file| avro::Reader::new(file);
+        let (tx, rx) = std::sync::mpsc::sync_channel(self.max_increment_size as usize);
+        let tail = if fc.tail {
+            FileReadStyle::TailFollowFd
+        } else {
+            FileReadStyle::ReadOnce
+        };
+        let path = fc.path.clone();
+        std::thread::spawn(move || {
+            read_file_task(path, tx, None, tail, ctor);
+        });
+
+        Some(RtFileConnector { stream: rx })
+    }
+
+    #[allow(clippy::redundant_clone)]
     fn create_rt_file_connector(
         &self,
         _id: SourceInstanceId,
-        _fc: FileSourceConnector,
-    ) -> RtFileConnector {
-        error!("Timestamping is unsupported for file sources");
-        RtFileConnector {}
+        fc: FileSourceConnector,
+    ) -> Option<RtFileConnector<std::vec::Vec<u8>>> {
+        let ctor = |fi| Ok(std::io::BufReader::new(fi).split(b'\n'));
+        let (tx, rx) = std::sync::mpsc::sync_channel(self.max_increment_size as usize);
+        let tail = if fc.tail {
+            FileReadStyle::TailFollowFd
+        } else {
+            FileReadStyle::ReadOnce
+        };
+        let path = fc.path.clone();
+        std::thread::spawn(move || {
+            read_file_task(path, tx, None, tail, ctor);
+        });
+
+        Some(RtFileConnector { stream: rx })
     }
 
-    fn create_rt_kinesis_connector(
+    fn create_byo_ocf_connector(
```

Comment:
I don't see why not, though it would make the create source syntax very complex. I think that if we ever move towards supporting that flexibility, we would want to create two separate commands in the form of `CREATE CONSISTENCY foo` / `CREATE SOURCE bar with (consistency=foo)`